TwitchPlaysAssembly/Src/ComponentSolvers/Helpers/ComponentSolverFactory.cs

```diff
@@ -688,136 +687,135 @@ public static void AddModuleInformation(ModuleInformation info)
+				return new MorseCodeComponentSolver(module);
 
 			case ComponentTypeEnum.WireSequence:
-				return new WireSequenceComponentSolver(bombCommander, (WireSequenceComponent) bombComponent);
+				return new WireSequenceComponentSolver(module);
 
 			case ComponentTypeEnum.Password:
-				return new PasswordComponentSolver(bombCommander, (PasswordComponent) bombComponent);
+				return new PasswordComponentSolver(module);
 
 			case ComponentTypeEnum.Maze:
-				return new InvisibleWallsComponentSolver(bombCommander, (InvisibleWallsComponent) bombComponent);
+				return new InvisibleWallsComponentSolver(module);
 
 			case ComponentTypeEnum.WhosOnFirst:
-				return new WhosOnFirstComponentSolver(bombCommander, (WhosOnFirstComponent) bombComponent);
+				return new WhosOnFirstComponentSolver(module);
 
 			case ComponentTypeEnum.NeedyVentGas:
-				return new NeedyVentComponentSolver(bombCommander, (NeedyVentComponent) bombComponent);
+				return new NeedyVentComponentSolver(module);
 
 			case ComponentTypeEnum.NeedyCapacitor:
-				return new NeedyDischargeComponentSolver(bombCommander, (NeedyDischargeComponent) bombComponent);
+				return new NeedyDischargeComponentSolver(module);
 
 			case ComponentTypeEnum.NeedyKnob:
-				return new NeedyKnobComponentSolver(bombCommander, (NeedyKnobComponent) bombComponent);
+				return new NeedyKnobComponentSolver(module);
 
 			case ComponentTypeEnum.Mod:
-				KMBombModule solvableModule = bombComponent.GetComponent<KMBombModule>();
+				KMBombModule solvableModule = module.BombComponent.GetComponent<KMBombModule>();
 				try
 				{
-					return CreateModComponentSolver(bombCommander, bombComponent, solvableModule.ModuleType, solvableModule.ModuleDisplayName);
+					return CreateModComponentSolver(module, solvableModule.ModuleType, solvableModule.ModuleDisplayName);
 				}
-				catch
+				catch (Exception exc)
 				{
-					DebugLog("Failed to create a valid Component Solver for Bomb Module: {0}", solvableModule.ModuleDisplayName);
-					DebugLog("Using Fallback Component solver instead.");
+					DebugHelper.LogException(exc, "An exception has occurred while creating a mod component solver:");
+					DebugLog("Failed to create a valid Component Solver for Bomb Module: {0}. Using Fallback Component solver instead.", solvableModule.ModuleDisplayName);
 					LogAllComponentTypes(solvableModule);
 
-					return new UnsupportedModComponentSolver(bombCommander, bombComponent);
+					return new UnsupportedModComponentSolver(module);
 				}
 
 			case ComponentTypeEnum.NeedyMod:
-				KMNeedyModule needyModule = bombComponent.GetComponent<KMNeedyModule>();
+				KMNeedyModule needyModule = module.BombComponent.GetComponent<KMNeedyModule>();
 				try
 				{
-					return CreateModComponentSolver(bombCommander, bombComponent, needyModule.ModuleType, needyModule.ModuleDisplayName);
+					return CreateModComponentSolver(module, needyModule.ModuleType, needyModule.ModuleDisplayName);
 				}
 				catch
```

Comment:
We should catch and log the exception just like is done for regular modules above.